### PR TITLE
Pipeline core

### PR DESCRIPTION
**Pull request by gearama**
First pass at a core perf pipeline . runs one test. 

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by LarryOsterman — COMMENTED** [review]
Not seeing anything wrong, but I'll let the ES folks chime in.

**Comment by gearama** [comment]
@danieljurek for your viewing pleasure

File: eng/pipelines/templates/jobs/perf.yml
Comment (mikeharder) [inline eng/pipelines/templates/jobs/perf.yml:5]:
```suggestion
- name: Services
```

File: eng/pipelines/templates/jobs/perf.yml
Comment (mikeharder) [inline eng/pipelines/templates/jobs/perf.yml:32]:
```suggestion
    Language: cpp
```

File: sdk/core/perf.yml
Comment (mikeharder) [inline sdk/core/perf.yml:5]:
```suggestion
  default: 'source'
```

I believe C++ currently only supports testing the source code in the branch (not a published package).  If so, this parameter should only be `source` until published packages are supported.

**Review by mikeharder — APPROVED** [review]
Approved pending a few small changes

**Comment by mikeharder** [comment]
@gearama: I created a PR to add Cpp to pipeline "tools - perf-automation - tests" (which is used to validate perf changes in the tools repo against all languages):

https://github.com/Azure/azure-sdk-tools/pull/4217

The pipeline is passing.  Once this PR to cpp is merged, I will update the tools PR to point to cpp main, and merge it as well.

**Comment by danieljurek** [comment]
From looking at the perf run, I'm seeing that we're re-building a lot of libraries that should already be cached in vcpkg. Let's meet again early next week to make some adjustments to the pipeline so that it's pulling `vcpkg` down at the expected commitish and using the cache appropriately.

**Comment by mikeharder** [comment]
> From looking at the perf run, I'm seeing that we're re-building a lot of libraries that should already be cached in vcpkg. Let's meet again early next week to make some adjustments to the pipeline so that it's pulling `vcpkg` down at the expected commitish and using the cache appropriately.

I noticed this as well.  Anything we can do to speed up the C++ build in the perf pipelines will be really helpful to people running short tests.

**Comment by mikeharder** [comment]
@danieljurek: It's also fine with me if you want to merge this PR as-is to get something working, then approve the vcpkg caching in a second PR.

**Review by danieljurek — APPROVED** [review]
LGTM. Let's fix the vcpkg building work in another PR.

**Pull request merged**
